compiler/rustc_typeck/src/check/coercion.rs

```diff
@@ -941,6 +941,10 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             exprs.len()
         );
 
+        if prev_ty == new_ty {
```

Comment:
Comment here referencing #88097 and coercion of a closure to itself

Reply:
Got it; fixed now.